test/org/sagebionetworks/bridge/services/ParticipantServiceTest.java

```diff
@@ -626,6 +655,126 @@ public void notSettingStatusDoesntClearStatus() {
+        StudyParticipant participant = new StudyParticipant.Builder().copyOf(PARTICIPANT)
+                .withEmail("updated@email.com").withEmailVerified(Boolean.FALSE).build();
+        
+        participantService.updateParticipant(STUDY, ImmutableSet.of(), participant);
+        
+        verify(accountDao).updateAccount(accountCaptor.capture());
+        Account account = accountCaptor.getValue();
+        verify(account).setEmail("updated@email.com");
+        verify(account).setEmailVerified(Boolean.FALSE);
+        verify(accountWorkflowService).sendEmailVerificationToken(STUDY, ID, "updated@email.com");
+    }
+
+    @Test
+    public void updateParticipantEmailCanBeAddedNoVerification() {
+        STUDY.setEmailVerificationEnabled(false);
+        mockHealthCodeAndAccountRetrieval(null, PHONE);
+        
+        StudyParticipant participant = new StudyParticipant.Builder().copyOf(PARTICIPANT)
+                .withEmail("updated@email.com").withEmailVerified(Boolean.FALSE).build();
+        
+        participantService.updateParticipant(STUDY, ImmutableSet.of(), participant);
+        
+        verify(accountDao).updateAccount(accountCaptor.capture());
+        Account account = accountCaptor.getValue();
+        verify(account).setEmail("updated@email.com");
+        verify(account).setEmailVerified(Boolean.FALSE);
+        verify(accountWorkflowService, never()).sendEmailVerificationToken(eq(STUDY), eq(ID), anyString());
+    }
+    
+    @Test
+    public void updateParticipantEmailCannotBeDeleted() {
+        STUDY.setEmailVerificationEnabled(true);
+        mockHealthCodeAndAccountRetrieval(EMAIL, null);
+        
+        StudyParticipant noEmail = new StudyParticipant.Builder().copyOf(PARTICIPANT)
+                .withEmail(null).build();
+        
+        participantService.updateParticipant(STUDY, ImmutableSet.of(), noEmail);
+        
+        // You can't delete email, so we are going to completely ignore it
+        verify(accountDao).updateAccount(accountCaptor.capture());
+        Account account = accountCaptor.getValue();
+        verify(account, never()).setEmail(anyString());
+        verify(account, never()).setEmailVerified(anyBoolean());
+        verify(accountWorkflowService, never()).sendEmailVerificationToken(eq(STUDY), eq(ID), anyString());
+    }
+    
+    @Test
+    public void updateParticipantPhoneCannotBeChanged() {
+        mockHealthCodeAndAccountRetrieval(EMAIL, PHONE);
+        when(account.getPhoneVerified()).thenReturn(Boolean.TRUE);
+        
+        StudyParticipant participant = new StudyParticipant.Builder().copyOf(PARTICIPANT)
+                .withPhone(OTHER_PHONE).withEmailVerified(Boolean.FALSE).build();
+        
+        participantService.updateParticipant(STUDY, ImmutableSet.of(), participant);
+        
+        verify(accountDao).updateAccount(accountCaptor.capture());
+        Account account = accountCaptor.getValue();
+        verify(account, never()).setPhone(PHONE);
```

Comment:
Should this be setPhone(any()) to make sure we're not setting this to any other phone number?

Reply:
Yes it seems any case of using never() should be paired with any(). Did a search and replace of a few more in this test.